homeassistant/helpers/entity_platform.py

```diff
@@ -433,8 +433,8 @@ def add_entities(
         if already_exists:
             msg = f"Entity id already exists: {entity.entity_id}"
             if entity.unique_id is not None:
-                msg += f". Platform {self.platform_name} does not generate unique IDs"
-            raise HomeAssistantError(msg)
+                msg += f". Platform {self.platform_name} does not generate unique IDs - ignoring entity"
+            return
```

Comment:
We need to log the error if we remove the raise.

I'm not sure we want to remove the raise here though.

Reply:
I agree that we need the log. we can either raise the exception or not, but if we do, it should be caught somewhere, otherwise why do we raise it.
Also, there are other exceptions there that have to be caught, so the catch in bootstrap.py is necessary anyway. Only question is what we do when there is a duplicate unique_id. You guys understand the architecture much better than me, so let me know what you prefer. In any case, it will be a minor change, but i need to know what the optimal expected behavior is.
Thanks

Reply:
anyway, pushed a commit that adds the log
please tell me if you want the exception back. It won't really change the behavior since no one was catching the exception anyway so it will be caught in the component async_setup and all other entities will be added (if i understand the behavior correctly)

Reply:
I think that not raising is ok. The raising was only breaking the entity component, who would then ignore it. I guess we could consider raising an error and then having the entity component unloading that platform? That way we punish platforms harder if they violate the rules.

Reply:
couldn't this introduce more instability that what it saves? i am no expert in hass, but unless we have many cases of components not returning unique_ids, you have to ask whether this is really helping.
out of curiosity, where would we unload the component? is that entity_component.py?

Reply:
We never unload components. We do reset components if we are reloading configuration. Platforms can be unloaded if the config entry is unloaded.

Reply:
Unloading a platform would mainly be a clear indicator to the user that the platform is broken and that the author should fix it. Let's stick to logging for now.